### PR TITLE
fix: check version before alter region

### PR DESCRIPTION
**Pull request by WenyXu**
I hereby agree to the terms of the [GreptimeDB CLA](https://gist.github.com/xtang/6378857777706e568c1949c7578592cc)

## What's changed and what's your intention?

check version before alter region

## Checklist

- [ ]  I have written the necessary rustdoc comments.
- [ ]  I have added the necessary unit tests and integration tests.

### CODE REVIEW

**Review by killme2008 — APPROVED** [review]

File: src/mito2/src/worker/handle_alter.rs [hunk 1]
```diff
@@ -47,6 +47,11 @@ impl<S> RegionWorkerLoop<S> {
 
         // Get the version before alter.
         let version = region.version();
+        if version.metadata.schema_version >= request.schema_version {
+            // Returns if it altered.
+            sender.send(Ok(Output::AffectedRows(0)));
```
Comment (killme2008) [inline src/mito2/src/worker/handle_alter.rs:52]: Adds a warning log for debugging purposes.

File: src/mito2/src/worker/handle_alter.rs [hunk 2]
```diff
@@ -47,6 +47,15 @@ impl<S> RegionWorkerLoop<S> {
 
         // Get the version before alter.
         let version = region.version();
+        if version.metadata.schema_version >= request.schema_version {
+            warn!(
+                "region schema version {} greater than request schema version {}",
```
Comment (killme2008) [inline src/mito2/src/worker/handle_alter.rs:52]: Adds the region id or name?

File: src/mito2/src/worker/handle_alter.rs [hunk 3]
```diff
@@ -47,6 +47,15 @@ impl<S> RegionWorkerLoop<S> {
 
         // Get the version before alter.
         let version = region.version();
+        if version.metadata.schema_version > request.schema_version {
+            warn!(
+                "region id:{}, region schema version {} greater than request schema version {}",
```
Comment (killme2008) [inline src/mito2/src/worker/handle_alter.rs:52]:
```suggestion
                "Ignored alert request, region id:{}, region schema version {} is greater than request schema version {}",
```

File: src/mito2/src/worker/handle_alter.rs [hunk 4]
```diff
@@ -47,6 +47,15 @@ impl<S> RegionWorkerLoop<S> {
 
         // Get the version before alter.
         let version = region.version();
+        if version.metadata.schema_version > request.schema_version {
```
Comment (evenyag) [inline src/mito2/src/worker/handle_alter.rs:50]: We check the schema version here
https://github.com/GreptimeTeam/greptimedb/blob/14e6998d4160b9f6d1fe4fd0a9cb27bb13a92c9f/src/store-api/src/region_request.rs#L191-L200

**Review by evenyag — APPROVED** [review]

**Review by fengjiachun — APPROVED** [review]

**Pull request merged**